router/index.js

```diff
@@ -24,20 +31,102 @@ router.post('/login', (req, res, next) => {
 
 // get list of cameras
 
-router.get('/camera', (req, res, next) => {
-  res.sendStatus(200);
+router.get('/camera', async (req, res, next) => {
+
+  try {
+    const allcameras = await db.getAllCameras();
+
+    if (allcameras.length === 0) {
+      res.json({message: 'no cameras available'});
+    } else {
+      res.send(allcameras);
+    }
+
+
+  } catch (error) {
+    next(error);
+    return; // Ensures rest of function doesn't run
+  }
 });
 
 // get camera by ID
 
-router.get('/camera/:id', (req, res, next) => {
-  res.sendStatus(200);
+router.get('/camera/:id', async (req, res, next) => {
+  // check if id is an integer
+  if (!Number.isInteger(parseInt(req.params.id))) {
+    res
+      .status(400)
+      .json({message:'ID needs to be an integer'})
+  } else {
+    db.getCameraById(req.params.id)
+      .then((camera) => {
+        if (camera == null) {
+          res
+            .status(400)
+            .json({message:'no camera found by ID specified'})
+        } else {
+          res.send(camera);
+        }
+      })
+      .catch(next);
+
+  }
 });
 
 // create camera
 
-router.post('/camera', (req, res, next) => {
-  res.sendStatus(200);
+router.post('/camera', async (req, res, next) => {
+
+  const jsonReqErrors = {};
+
+  if (!req.body.brand) {
+    jsonReqErrors.brand = 'Error: json key/value pair of brand not defined or empty. Required';
+  }
+  if (!req.body.model) {
+     jsonReqErrors.model = 'Error: json key/value pair of model not defined or empty. Required';
+  }
+  if (!req.body.sensor) {
+    jsonReqErrors.sensor = 'Error: json key/value pair of sensor not defined or empty. Required';
+  }
+  if (!req.body.megapixels) {
+    jsonReqErrors.megapixels = 'Error: json key/value pair of megapixels not defined or empty. Required';
+  }
+  if (!req.body.viewfinder) {
+    jsonReqErrors.viewfinder = 'Error: json key/value pair of viewfinder not defined or empty. Required';
+  }
+  if (!req.body.price) {
+    jsonReqErrors.price = 'Error: json key/value pair of price not defined or empty. Required';
+  }
+
```

Comment:
This big if chain works but it could be made smaller if you want.
Use Object.keys(req.body) to get an array of  keys in the request body. After that you can use a forEach to check if the given keys match a required list of keys defined by you.

This will save you from writing essentially the same thing for each key. and help you scale if suddenly you have a lot more required keys in the future.

Reply:
Good idea. I will explore your idea and made the code simpler to understand